tests/unit/jobserver/views/test_projects.py

```diff
@@ -200,6 +200,32 @@ def get_repo_is_private(self, *args):
     assert "Public" not in response.rendered_content
 
 
+def test_projectdetail_with_timed_out_github(rf):
+    project = ProjectFactory()
+    WorkspaceFactory(
+        project=project, repo=RepoFactory(url="https://github.com/owner/repo")
+    )
+
+    request = rf.get("/")
+    request.user = UserFactory()
+
+    class BrokenGitHubAPI:
+        def get_repo_is_private(self, *args):
+            raise TimeoutError
+
+    response = ProjectDetail.as_view(get_github_api=BrokenGitHubAPI)(
+        request, project_slug=project.slug
+    )
+
+    assert response.status_code == 200
+
+    # check there is no public/private badge when GitHub returns an
+    # unsuccessful response
+    assert not response.context_data["public_repos"]
+    assert response.context_data["private_repos"][0]["is_private"] is None
+    assert "Public" not in response.rendered_content
```

Comment:
`response.rendered_context` will call `response.render()`, which will render `templates/project/detail.html`. Alas, this template has a missing template variable. I'm going to guess that it's `Project.org`, so try the following...

```py
project = ProjectFactory(org=OrgFactory())
```

Reply:
This only started failing once I rebased, I wonder what changed such that the tests passed previously. Either way, thanks for the hint - I'll give it a go!

Reply:
I reckon you branched and rebased either side of #4505.